contract_compassion/tests/test_contract_compassion.py

```diff
@@ -154,3 +154,31 @@ def test_reset_open_invoices(self):
         self.assertEqual(invoice_line_up.price_unit, contract_line.amount)
         self.assertEqual(
             invoice_line_up.price_subtotal, contract_line.subtotal)
+
+    def test_contract_compassion_third_scenario(self):
+        contract_group = self._create_group(
+            'do_nothing', self.partners.ids[0], 1,
+            self.payment_term_id,
+            other_vals={'recurring_value': 1, 'recurring_unit': 'month'})
+        contract_group2 = self._create_group(
+            'do_nothing', self.partners.ids[1], 1,
+            self.payment_term_id,
+            other_vals={'recurring_value': 1, 'recurring_unit': 'month'})
+
+        contract = self._create_contract(
+            datetime.today().strftime(DF), contract_group,
+            datetime.today().strftime(DF),
+            other_vals={'channel': 'postal', 'type': 'O'})
+        contract_group.write({'partner_id': self.partners.ids[1]})
+        contract_group.on_change_partner_id()
+        self.assertTrue(contract_group.bvr_reference)

```

Comment:
You could add an assert that the bvr_reference is not the same as the one before you changed the partner of the contract.